fastjet/internal/delaunay/triangulation.go

```diff
@@ -0,0 +1,663 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+)
+
+// holds necessary information for the
+// delaunay triangulation
+type Delaunay struct {
+	triangles []*Triangle // all triangles created
+	root      *Triangle   // triangle that contains all points. Used to find a triangle
+	maxX      float64
+	minX      float64
+	maxY      float64
+	minY      float64
+}
+
+// creates a delaunay triangulation with the given points
+// all points have to be inside the user defined bounds
+func NewDelaunay(points []*Point, maxX float64, minX float64, maxY float64, minY float64) *Delaunay {
+	// root Triangle is a triangle that contains all points
+	dx := maxX - minX
+	dy := maxY - minY
+	a := NewPoint(minX-8*dx, maxY+10, -1)
+	b := NewPoint(maxX+8*dx, maxY+10, -1)
+	c := NewPoint(minX+dx/2, minY-8*dy, -1)
+	root := NewTriangle(a, b, c)
+	d := &Delaunay{
+		root: root,
+		maxX: maxX,
+		minX: minX,
+		maxY: maxY,
+		minY: minY,
+	}
+	d.triangles = make([]*Triangle, 0)
+	for _, p := range points {
+		d.InsertPoint(p)
+	}
+	return d
+}
+
+func (d *Delaunay) GetTriangles() []*Triangle {
```

Comment:
`func (d *Delaunay) Triangles() []*Triangle {`; the Go convention is that getters are named for the thing they get, setters are named `Set<Thing>`.